cql3/restrictions/statement_restrictions.cc

```diff
@@ -504,7 +504,6 @@ bool statement_restrictions::need_filtering() const {
         number_of_filtering_restrictions += _clustering_columns_restrictions->size() - _clustering_columns_restrictions->prefix_size();
     }
     return number_of_restricted_columns_for_indexing > 1
-            || (number_of_restricted_columns_for_indexing == 0 && _partition_key_restrictions->empty() && !_clustering_columns_restrictions->empty())
```

Comment:
What if the clustering column restrictions are not in order?

`SELECT * FROM tab WHERE ck2 = ?` (ck1 unrestricted)?

Reply:
Handled by line 504.  We have tests for it.

Reply:
Well, aside everything I stated in my previous reply, I am definitely in favor of simplifying this abomination of a condition... which is yet another argument for going ahead and merging this patch.